### ISSUE TITLE
`ToNativeExact t ρ` as type family

### ISSUE THREAD

**Issue by nomeata** [target/row-types#51]
I am playing around with `row-types`; neat work.

I want to connect it to some other generic library (not shared yet), but I am facing a wrinkle. `ToNativeExact t ρ` is a type class. I believe I could really use a type family that goes from the generic `Rep t` type to a `Row r`. Would that be possible?

**Comment by dwincort** [comment]
Not only would it be possible, but I've already been playing around with it: https://github.com/target/row-types/pull/50!  That said, I've been a little hesitant to merge that branch for two reasons:

A. I added a sub-library for aeson support, but that means it requires Cabal >=3.0.

B. It's a breaking change because I renamed the native functions slightly: `toNativeExact` became `toNative` and the old `toNative` became `toNativeGeneral`.  As I say in the PR, it feels more natural to me that `toNative`/`fromNative` are inverses.

I don't really have any co-contributors on row-types right now, and I don't have a good sense of who uses it, so since you clearly want to use it, I'll ask you: which of the following three options would you like to see:

1. Merge https://github.com/target/row-types/pull/50 basically as is.

2. Split out the aeson sub-library and just merge the native stuff.

3. Pull out _just_ the `NativeRow` type family as its own PR, treat it as a minor version bump, and punt on the rest of this until later.

In short, do you think the renaming makes sense and/or would the aeson sub-library be problematic for you?

**Comment by nomeata** [comment]
I’m still playing around with it, so I probably am not authorative. Maybe see if these packages use the type class: https://packdeps.haskellers.com/reverse/row-types

Sublibraries may hinder adoption, so maybe 2 is reasonable?

**Comment by dwincort** [comment]
Cool!  I did not know about the reverse package search, but that is a neat tool.

It looks like there's nothing published with `native` functions, so I feel pretty good about option 2.  I'll leave the aeson stuff on the backburner.  I'll get to it asap.

**Comment by dwincort** [comment]
There is now a `NativeRow` type family that I think does what you want.  Check out https://hackage.haskell.org/package/row-types-0.4.0.0.
